### PR TITLE
Fix null pointer constant in HessianApprox

### PR DESCRIPTION
**Pull request by rscircus**
A whole class was assigned `0`.

### CODE REVIEW

File: src/main.cpp [hunk 1]
```diff
@@ -149,7 +149,7 @@ int main(int argc, char *argv[]) {
          ndesign_global);
 
   /* Initialize Hessian approximation */
-  HessianApprox *hessian = 0;
+  HessianApprox *hessian;
```
Comment (steffi7574) [inline src/main.cpp:152]: I guess it'd be a good idea to set this pointer to NULL? Otherwise it might be unassigned after the switch...

Comment (rscircus) [reply src/main.cpp:152]: That's why I added the `default` case to the switch clause. Just read on. ;)

Here is what you have to look for:
```
    default:
      printf("Error: unexpected hessianapprox_type returned");
      return 0;
  }
```
Further `0 == NULL` evaluates to false. `NULL` (non legal address) is not `0` (an integer value). Sometimes `0` is overridden by libraries to be a *null pointer constant*, but I prefer the solution above as it is more clear and does not have any hidden assumptions.

Comment (rscircus) [reply src/main.cpp]: Otherwise your point is completely valid. :)

Comment (rscircus) [reply src/main.cpp:152]: Otherwise your point(er) is completely valid. :)

Comment (steffi7574) [reply src/main.cpp:152]: OK, sorry didn't see that. Let's still set it to NULL then. Shoudn't hurt...

**Pull request closed without merging**